### DISCUSSION TITLE
ImportError: no module named 'socket'

### DISCUSSION THREAD

**Discussion by ElToro1966** [micropython/micropython#9762] [Libraries & Drivers] [answered]
I am trying to get an mqtt client working on a Raspberry Pico H with the umqtt.simple library. When I try to import MQTTClient from the library,
`from umqtt.simple import MQTTClient`
I get the following error on running the code:
```
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
  File "/lib/umqtt/simple.py", line 1, in <module>
  File "/lib/usocket.py", line 2, in <module>
ImportError: no module named 'socket'
```
When I go to the packet manager in Thonny, I find no socket library. However, the library is supposed to be a part of the [standard packages in micropython](https://docs.micropython.org/en/v1.18/library/socket.html?highlight=socket#module-socket). 

I am running the latest uf2-file for Pico, that is 1.19.1 (rp2-pico-20220618-v1.19.1.uf2).

Help much appreciated.

**Comment by robert-hh** [comment, marked as answer]
For the RPi pico.W you have to use the matching firmware. The firmware for the standard RPi pico does not support networking. You may also try the most recent daily build.

**Reply by ElToro1966** [reply]
> For the RPi pico.W you have to use the matching firmware. The firmware for the standard RPi pico does not support networking. You may also try the most recent daily build.

Sorry, I forgot to mention that I am using a Pico H with a ThingsOnEdge Cricket, so there is networking, but no on the Pico itself. Should I try to install the Pico W firmware on the Pico H to get access to networking?

**Reply by robert-hh** [reply]
That will not work, since there is no driver for that wifi module in the firmware.
